mlibrary/search
# [LIBSEARCH-899] Convert ChooseAffiliation into a select dropdown filter option.

# Overview
The modal used for the `ChooseAffiliation` component is unsupported by versions of Safari older than 15.4, rendering the site unusable. The component has been redesigned to become a dropdown option under `Filters`, similar to what is seen for `Library Scope`, removing the need for a modal.

This pull request resolves [LIBSEARCH-899](https://mlit.atlassian.net/browse/LIBSEARCH-899).

## Anything else?
The new component no longer relies on cookies, and instead uses `localStorage`. This removed the need for the `affiliationCookieSetter` function and the `react-cookie` package.

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Open the [local site](http://localhost:3000/everything).
  - Look around the site and check to see if anything is broken.
  - Do you no longer see the `Ann Arbor|Flint` toggle in the website header?
  - Make a search in one of the datastores
    - Do you see `Affiliation` at the bottom of the filter options?
    - Is it set to one of the schools?
    - If you click on the other affiliation, does it reload the page?
    - If you click on the current affiliation, does the page not reload?
    - If you close the tab and reopen it, is it set to the affiliation you last selected?
    - Compare the same search to the live site. Do the same options still appear?
